dot/network/notifications.go

```diff
@@ -247,6 +242,15 @@ func (s *Service) broadcastExcluding(info *notificationsProtocol, excluding peer
 			logger.Trace("sending handshake", "protocol", info.protocolID, "peer", peer, "message", hs)
 			err = s.host.send(peer, info.protocolID, hs)
 		} else {
+			if s.host.messageCache != nil {
+				var added bool
+				added, err = s.host.messageCache.Put(peer, msg.String())
```

Comment:
this should use `msg.Encode()` instead of `msg.String()`, not sure if `String` will correctly represent all the fields

Reply:
Done